packages/ui-source-code-editor/src/SourceCodeEditor/props.ts

```diff
@@ -0,0 +1,292 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import PropTypes from 'prop-types'
+
+import type { TagStyle } from '@codemirror/language'
+
+import { controllable } from '@instructure/ui-prop-types'
+
+import type {
+  PropValidators,
+  CodeEditorTheme,
+  OtherHTMLAttributes
+} from '@instructure/shared-types'
+import type {
+  WithStyleProps,
+  StyleObject,
+  ComponentStyle
+} from '@instructure/emotion'
+import type { WithDeterministicIdProps } from '@instructure/ui-react-utils'
+import type { BidirectionalProps } from '@instructure/ui-i18n'
+
+type SourceCodeEditorOwnProps = {
+  /**
+   * The label text that screen readers will read when this component gets
+   * focus.
+   */
+  label: string
+
+  /**
+   * The language to use. When not given, this will default to the first
+   * language that was loaded.
```

Comment:
How do you determine what language was loaded first? Can users influence it? I think a bit more explanation would be nice here.

Reply:
Oh, that was copied from the old code editor, I modified it! Thanks!